external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaRecordToUnsafeRowConverter.scala

```diff
@@ -30,13 +30,19 @@ private[kafka010] class KafkaRecordToUnsafeRowConverter {
 
   def toUnsafeRow(record: ConsumerRecord[Array[Byte], Array[Byte]]): UnsafeRow = {
     rowWriter.reset()
+    rowWriter.unsetNullAt(0)
+    rowWriter.unsetNullAt(1)
```

Comment:
Here we unset null in `rowWriter` to avoid missing non-null value  in record key and record value. New unit test covers this case.

Reply:
If `unset` is missed, `rowWriter.reset()` is the better place to `unset` them. Then, we don't need to add https://github.com/apache/spark/pull/24441/files#diff-c654dda59109239bce5498cbc30ab251R114 in this PR.

cc @cloud-fan , @zsxwing

Reply:
After `reset`, we should also call `zeroOutNullBytes`

Reply:
@cloud-fan +1, i missed this method